src/NuGet.Core/NuGet.Packaging/Rules/NoRefOrLibFolderInPackageRule.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Client;
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using NuGet.RuntimeModel;
+
+namespace NuGet.Packaging.Rules
+{
+    class NoRefOrLibFolderInPackageRule : IPackageRule
+    {
+        private static readonly string BuildDir = PackagingConstants.Folders.Build + Path.DirectorySeparatorChar;
+        public string MessageFormat { get; }
+
+        public NoRefOrLibFolderInPackageRule(string messageFormat)
+        {
+            MessageFormat = messageFormat;
+        }
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader package)
+        {
+            var files = package.GetFiles().ToList();
```

Comment:
no need to to `.ToList()` if you only use it once and it gets passed to a method that takes an `IEnumerable`